SparkPost/matchbox
# UX-222 Modal component update

### What Changed

- Old Modal component moved to `Modal.LEGACY`
- New `Modal` component
- New `Modal.Header` Component
- New `Modal.Content` Component
- New `Modal.Footer` Component

### How To Test or Verify

- `npm run start:storybook`
- Visit the Modal story and verify the modal looks right and the API is what is expected

### PR Checklist
